### PR TITLE
Add model signing as a sandbox project

### PR DESCRIPTION
**Pull request by mihaimaruseac**
We have a working group that meets for model signing work, as part of https://github.com/ossf/ai-ml-security/issues/10. Since this working group helps in developing https://github.com/sigstore/model-transparency and building standards around it, it needs to be officially a project, not a WG.

We add the project at a sandbox stage.

Please note that there are 2 repos invovled here:

- https://github.com/sigstore/model-transparency which will be owned by Sigstore and is just the implementation work for the library for model signing
- a new repository to be created under https://github.com/ossf to represent standard documents, as outputs of this project

These two repositories should work in unison to achieve a common goal.

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
changing to draft as there needs to be some work regarding repo

**Comment by lehors** [comment]
I think the naming needs some work.

First the repo is called model_transparency but the proposed project name is "Model Signing". Then looking into the repo README it appears that model signing is only part of what's being done (i.e., Model Signing + SLSA).

Finally, "Model Transparency" is actually a complex topic which is quite broader than what this project aims to tackle. See [The Foundation Model Transparency Index](https://arxiv.org/abs/2310.12941).

So, while I'm not really sure what name should be used I think this ought to be sorted out. We should find another name and rename the repo accordingly to get everything aligned.

**Comment by mihaimaruseac** [comment]
Yes, the repo confusion is why I made this as a draft for now, apologies for the notifications.

The sigstore/model_transparency repo and the SIG (to become project) repo should be different entities, since they had different histories. I'll update this PR with better naming and documentation, pending some external discussions.

**Review by SecurityCRob — COMMENTED** [review]
please get the names synchronized.  the effort has merit, lets make sure it is clearly documented and communicated

**Comment by marcelamelara** [comment]
@mihaimaruseac Bumping this PR. Please let us know if/how the TAC can help the Project sort things out.

**Comment by mihaimaruseac** [comment]
Sorry, got delayed with a bunch of other items and missed this one. Will do by the end of the week. Thank you for the bundge

**Comment by mihaimaruseac** [comment]
Made it clearer that we have 2 separate repositories, under 2 separate organizations. The model signing SIG (to become project) under the AI/ML WG will have a repo under OpenSSF for the work regarding standardizations, efficiency, etc., but the technical signing work that is developed as part of sigstore/model-transparency will stay with Sigstore, since it encompasses more than what the SIG (to become project) aims to do.

Force pushed an amended commit and marked the PR as ready for review. Apologies it took so long to do this.

**Review by SecurityCRob — APPROVED** [review]

**Review by mlieberman85 — APPROVED** [review]
LGTM

**Review by sevansdell — APPROVED** [review]
Great work!

**Review by bobcallaway — APPROVED** [review]

**Review by steiza — APPROVED** [review]

File: process/project-lifecycle-documents/model_signing_sandbox_stage.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+> held within a directory. This specification may have wider applicability to
+> signing directories of multiple arbitrary file formats. This specification may
+> later be proposed as a formal standard.
+
+#### Specific Goals Include:
+
+* Develop standards for efficient hashing of large models
+* Develop standards for efficient verification of models that contain multiple
+  formats in the same place
+* Develop standards for efficient verification of models at inference time.
```
Comment (marcelamelara) [inline process/project-lifecycle-documents/model_signing_sandbox_stage.md:31]: Just to be clear, by verification here, we mean hash-based integrity checking? Verification is a pretty broad term, so if the scope for this project is narrower, I think we should clarify that in the goals.

File: process/project-lifecycle-documents/model_signing_sandbox_stage.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+To achieve these goals, we work on developing
+https://github.com/sigstore/model-transparency library as an OSS standard for ML
+models signing and verification. The aim of this project is to guide the
+`model-transparency` development and help in standardizing hashing,
+verification, and deployment (e.g., model signature format).
+
+#### Non-Goals Include:
+
+* Developing a new model format
+* Handling security of ML outputs
```
Comment (marcelamelara) [inline process/project-lifecycle-documents/model_signing_sandbox_stage.md:42]: I'm assuming Assertions/verification of training data quality/bias/etc. or the training pipeline are outside of the scope? If there are other major non-goals, I suggest adding those, too.

File: process/project-lifecycle-documents/model_signing_sandbox_stage.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    part of Sigstore, which is already an OpenSSF (thus, LF) member
+  * Standardization work, etc. will occur under a new repository to be created
+    under OpenSSF
+
+### Project References
+
+The project should provide a list of existing resources with links to the repository, and if available, website, a roadmap, contributing guide, demos and walkthroughs, and any other material to showcase the existing breadth, maturity, and direction of the project.
+
+| Reference          | URL |
+|--------------------|-----|
```
Comment (marcelamelara) [inline process/project-lifecycle-documents/model_signing_sandbox_stage.md:91]: Though it's not a requirement for sandbox entry, I think a row for the Meeting agenda should be added with a link to the meeting notes.

**Review by marcelamelara — CHANGES_REQUESTED** [review]
Thanks for the updates @mihaimaruseac ! I have a couple of suggestions related to making the scope of the project clearer, but otherwise, I think this application is almost ready to go.

File: process/project-lifecycle-documents/model_signing_sandbox_stage.md (hunk 3 above)
Comment (mihaimaruseac) [reply process/project-lifecycle-documents/model_signing_sandbox_stage.md:91]: Thank you! This is a good idea, I added a link to this year's notes.

File: process/project-lifecycle-documents/model_signing_sandbox_stage.md (hunk 2 above)
Comment (mihaimaruseac) [reply process/project-lifecycle-documents/model_signing_sandbox_stage.md:42]: Great idea, thank you! Added this as a bullet and made it explicit that anything outside of model integrity is out of scope

File: process/project-lifecycle-documents/model_signing_sandbox_stage.md (hunk 1 above)
Comment (mihaimaruseac) [reply process/project-lifecycle-documents/model_signing_sandbox_stage.md:31]: Thank you for the clarification! Updated to make it clear that both here and in line 29 we only mean integrity of the model bytes.

**Comment by mihaimaruseac** [comment]
Thank you for the reviews! Updated and rebased on latest `main`

**Review by marcelamelara — APPROVED** [review]
Thanks for the updates! LGTM.

**Pull request merged**